src/System.Private.CoreLib/src/System/StubHelpers.cs

```diff
@@ -1715,14 +1718,18 @@ internal static void CheckStringLength(int length)
 
         internal static void CheckStringLength(uint length)
         {
-            if (length > 0x7ffffff0)
+            const int MaxStringLength = 0x7ffffff0;
```

Comment:
Nit: I would leave the constant unnamed as before - so that this does not get confused with maximum managed string length that is like half of this.